src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -140,22 +140,28 @@ def and_constraints(parsed_constraints: Sequence[Requirement]) -> Requirement:
             merged_specs: Set[Tuple[str, str]] = set()
             expected_interpreter = parsed_constraints[0].project_name
             for parsed_constraint in parsed_constraints:
-                if parsed_constraint.project_name != expected_interpreter:
```

Comment:
FYI the below code is to create an error message when you try to AND interpreter constraints with different interpreter names, like `PyPY AND CPython`.